storage/storage.go

```diff
@@ -535,13 +535,12 @@ func v4SanitizeHeaders(hdrs []string) []string {
 		sanitizedHeader := strings.TrimSpace(hdr)
 
 		var key, value string
-		headerMatches := strings.Split(sanitizedHeader, ":")
-		if len(headerMatches) < 2 {
+		headerMatches := canonicalHeaderRegexp.FindStringSubmatch(sanitizedHeader)
```

Comment:
I think not using the canonicalHeaderRegexp is actually an intentional difference between v2 and v4; see comments in L526-527.

However, maybe we should use [strings.SplitN](https://pkg.go.dev/strings#SplitN) instead so that only the first `:` is split on. Does that sound like a good solution to you?